### PR TITLE
Add rubocop

### PR DESCRIPTION
**Pull request by brebory**
Adds rubocop and hound config for enforcing code style guide in PRs.

### CODE REVIEW

File: spec/type_spec.rb [hunk 1]
```diff
@@ -1,66 +1,67 @@
+# rubocop:disable Metrics/BlockLength
```
Comment (brebory) [inline spec/type_spec.rb]: Resolved

Comment (Beznus) [inline spec/type_spec.rb]: why disable this?

Comment (Beznus) [inline spec/type_spec.rb]: oh I see where you are disabling this

Comment (brebory) [inline spec/type_spec.rb]: Yeah, just in specs - they can get really long and I don't see any other way of getting around it besides disabling it in offending spec files.

**Pull request merged**